src/OmniSharp.DotNetTest/VSTestManager.cs

```diff
@@ -309,6 +317,32 @@ public override async Task<RunTestResponse> RunTestsInContextAsync(int line, int
                 node = node.Parent;
             }
 
+            return (methodNames, testFramework?.Name);
+
+            static bool isTestMethod(IMethodSymbol methodSymbol, ref TestFramework? framework)
+            {
+                if (framework is object)
+                {
+                    return framework.IsTestMethod(methodSymbol);
+                }
+
+                foreach (var f in TestFramework.Frameworks)
```

Comment:
Technically, this means that you can't combine multiple testing framework types in a single file and have them run all at once. I think that's probably fine.